src/main/java/io/quarkus/search/app/hibernate/InputProvider.java

```diff
@@ -1,10 +1,93 @@
 package io.quarkus.search.app.hibernate;
 
+import java.io.FileInputStream;
 import java.io.IOException;
 import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
 
-public interface InputProvider {
+import io.quarkus.search.app.util.CloseableDirectory;
 
-    InputStream open() throws IOException;
+import io.quarkus.logging.Log;
 
+import org.jsoup.nodes.Document;
+import org.jsoup.nodes.Element;
+
+public record InputProvider(Path content) {
+
+    public InputStream open() throws IOException {
+        return new FileInputStream(content.toFile());
+    }
+
+    public static InputProvider from(Document document, CloseableDirectory temp, Object context) {
```

Comment:
It seems to me this is just moving parsing to an earlier phase of indexing, and adding an extra file write/read (for the temporary document), so I'm not sure how exactly this would help with memory consumption... but if you tested locally and it did, well, that's good news :shrug:

Reply:
it did help with the local runs. Yeah this moves it to a different phase, but the difference is that here we would only go through `read/parse with jsoup/ extract content / encode content` once rather than doing it  3 times (for one language) when creating the document (we have 3 fields using the input provider). So, that should reduce the memory consumption a bit. But I agree that we probably would want to return to this at some point to look closer into it.